packages/opentelemetry-plugin-grpc/src/utils.ts

```diff
@@ -48,3 +54,65 @@ export const _grpcStatusCodeToCanonicalCode = (
 export const _grpcStatusCodeToSpanStatus = (status: number): Status => {
   return { code: status };
 };
+
+/**
+ * Returns true if the metadata contains
+ * the opentelemetry outgoing request header.
+ */
+export const _containsOtelMetadata = (
+  metadata: grpcTypes.Metadata
+): boolean => {
+  return metadata.get(_otRequestHeader).length > 0;
+};
+
+/**
+ * Returns true if methodName matches pattern
+ * @param methodName the name of the method
+ * @param pattern Match pattern
+ */
+const _satisfiesPattern = (
+  methodName: string,
+  pattern: IgnoreMatcher
+): boolean => {
+  if (typeof pattern === 'string') {
+    return pattern.toLowerCase() === methodName.toLowerCase();
+  } else if (pattern instanceof RegExp) {
+    return pattern.test(methodName);
+  } else if (typeof pattern === 'function') {
+    return pattern(methodName);
+  } else {
+    throw new TypeError('Pattern is in unsupported datatype');
+  }
+};
+
+/**
+ * Returns true if the current plugin configuration
+ * ignores the given method.
+ * @param methodName the name of the method
+ * @param ignoredMethods a list of matching patterns
+ * @param onException an error handler for matching exceptions
+ */
+export const _methodIsIgnored = (
```

Comment:
this method should not throw anything, you can simply return boolean and make the same logic